src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/search/FuzzyScorer.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search
+
+import org.apache.commons.text.similarity.LevenshteinDistance
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Weights.Companion.FUZZY_CLOSE_MATCH
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Weights.Companion.FUZZY_SLIGHT_MATCH
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Weights.Companion.NO_MATCH
+
+class FuzzyScorer {
+  fun score(term: String, field: String): Int? {
+    if (field.isBlank() || term.isBlank()) {
+      return NO_MATCH
+    }
+    return when (term distanceTo field) {
+      -1 -> NO_MATCH
+      0 -> NO_MATCH
+      1 -> FUZZY_CLOSE_MATCH
+      else -> FUZZY_SLIGHT_MATCH
+    }
+  }
+
+  private infix fun String.distanceTo(field: String): Int {
+    // Short terms are less likely to have typos
+    val allowedDistance = when {
+      this.length < 3 -> 0
+      this.length < 6 -> 1
+      else -> 2
+    }
```

Comment:
it was quite easy to add the length specific behaviour that [elastic search implements](https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness)